### PR TITLE
DIV-4468 - remove the checkCcd flag

### PR DESCRIPTION
**Pull request by hosainnet**
# Description

We should always check CCD, to make AAT and prod like-for-like

Fixes #DIV-4468

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/petition/CcdRetrieveCaseTest.java [hunk 1]
```diff
@@ -119,66 +119,19 @@ public void givenMultipleAwaitingPaymentAndNoSubmittedCaseInCcd_whenRetrieveCase
...
-    @Test
-    public void givenDoNotCheckCcdAndOnePetitionInDivorceFormat_whenRetrieveCase_thenReturnPetition() throws Exception {
-        final String userToken = getUserToken();
-
-        final String caseInDivorceFormatFileName = DIVORCE_FORMAT_DRAFT_CONTEXT_PATH + "addresses.json";
-
-        createDraft(userToken, caseInDivorceFormatFileName,
-            Collections.singletonMap(DIVORCE_FORMAT_KEY, true));
-
-        Response cmsResponse = retrieveCase(userToken, false);
```
Comment (hosainnet) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/petition/CcdRetrieveCaseTest.java]: No longer relevant

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java [hunk 2]
```diff
@@ -51,15 +51,12 @@
     private UserService userService;
 
     @Override
-    public CaseDetails retrievePetition(String authorisation, Map<CaseStateGrouping, List<CaseState>> caseStateGrouping,
-                                        boolean checkCcd) throws DuplicateCaseException {
-        CaseDetails caseDetails = null;
+    public CaseDetails retrievePetition(String authorisation, Map<CaseStateGrouping, List<CaseState>> caseStateGrouping
+    ) throws DuplicateCaseException {
 
         Draft draft = draftService.getDraft(authorisation);
```
Comment (alexwakeman) [inline src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java]: I know this is unrelated to your work but wouldn't it be a better optimisation to wait until we know the CaseDetails case is null before checking the draft store?

Comment (hosainnet) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java]: Both `if` branches below are using the draft so I'm not sure we can optimise this further

Comment (alexwakeman) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java]: As discussed we are sharing this method across different controller routes, we could maybe raise a tech debt to sort this out properly, one for draft and one for Case, or to optimise the `retrievePetition` and `ccdRetrievalService.retrieveCase` logic, it's confusing and hard to maintain - plus nobody is sure if we really need it.

**Pull request merged**